### ISSUE TITLE
Support room upgrades

### ISSUE THREAD

**Issue by non-Jedi** [Nheko-Reborn/nheko#187]
**Is your feature request related to a problem? Please describe.**

I am in several rooms that have been upgraded. Showing the old rooms with tombstones frequently confuses me and causes me to click on the old room rather than the upgraded one.

**Describe the solution you'd like**

Support [room upgrades](https://matrix.org/docs/spec/client_server/latest#id160) by hiding rooms that have been `m.room.tombstone`d if the user is in `replacement_room`. It would also be nice to support joining `replacement_room` from the tombstoned room.

**Additional context**

In Riot, a tombstoned room has this instead of the message composition area:

![riot_tombstone_compose_area](https://user-images.githubusercontent.com/17915621/80616007-682a4f80-8a0e-11ea-9633-3b1f7f8d0560.png)

**Comment by deepbluev7** [comment]
Yes, this is on my to do list. This is currently blocked by #168, since I want to provide a better interface than Riot currently does. Just linking to the new room looks pretty bad imo, so I think we want to provide a seamless transition between rooms and kind of stitch them together. This is however a bit hard to do and will have to wait a bit, until we reworked our cache architecture a bit. mtxclient already supports tombstone events, Nheko is just missing the UI for it.

We could maybe add an intermediate step though, but we may also want to have room linkification working first.

**Comment by grinapo** [comment]
Right now riot isn't that bad: there are jump markers at the top and bottom of the message flow.
Since upgrade rooms are basically **new rooms** I don't think there's merit in stitching them together. (←opinion piece)

It's a problem, however, that tombstone events are unhandled now and whena  room have more events (message, leave, etc.) then the tombstone may scroll out at the top and it's almost impossible to see the room is no more.

**Comment by symphorien** [comment]
It's even possible to inadvertently add new messages to the old room despite the tombstone event...

**Comment by Hyask** [comment]
Is this still on somebody's radar?

**Comment by grinapo** [comment]
Oh yeah, on the radar. We're watching. :-) 6 years have gone by, amazing.
